js/kucoin.js

```diff
@@ -1801,6 +1801,14 @@ module.exports = class kucoin extends Exchange {
             const payload = timestamp + method + endpoint + endpart;
             const signature = this.hmac (this.encode (payload), this.encode (this.secret), 'sha256', 'base64');
             headers['KC-API-SIGN'] = this.decode (signature);
+            const partnerId = this.options['partnerId'];
```

Comment:
Hi! Thanks for the PR!

However, we have to add safe-guards, like we do with other options, for example, here:

- https://github.com/ccxt/ccxt/blob/master/js/kucoin.js#L1773

This is explained here: 

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sanitizing-input-with-safe-methods

If we leave it like that it will crash in Python and PHP.
Let me know if you can add those guards to it.
Thx again!

Reply:
Awesome, thanks for the quick reply! I added a new commit to the PR.

Reply:
@zholmes1 I've added a couple of edits, so, it should be used like so:

```
const exchange = new ccxt.kucoin({
    'options': {
        'partner': {
            'id': 'YOUR_PARTNER_ID',
            'secret': 'YOUR_PARTNER_SECRET',
        },
    },
})
```

Note that `partner` is now a dictionary in the options.
I will merge it shortly. Let me know if anything. Thanks again!